### DISCUSSION TITLE
Rolling over budget to next month?

### DISCUSSION THREAD

**Discussion by Mreco99** [actualbudget/releases#255] [Q&A]
Hi
if I have (lets say) £30 left over in last (April) months budget, im not sure what I should do with it. I dont want to over allocate it in April. 
I could just leave it? or
If I click on the amount 'To Budget' I get four options
Move to a category (self explanatory)
Hold for next month
Hold for all future months
Reset next months buffer

I dont actually know what the effect of those options will do, I dont want to select an option and it screws up my budget

If you can explain what those options will precisely do that would be great thanks
My choice would be to Move the left over budget from April to May, so that April says Zero
It makes sense that if I now go back and allocate money in April (retrospectively), it reduces all subsequent months budget, but if I leave the £30 there and now budget moneys in May, Aprils budget doesnt change. 

I believe that 'effectively' Aprils leftover budget has already rolled over, even if I do nothing with Aprils leftovers. Its just feels weird to scroll back in the budget and see moneys left in previous months

Im sure its just me not having an accountant brain. 

Thanks
Simon

**Comment by bpaulien** [comment]
- Move to a category (self explanatory) - yes, that's actually moving the money to a category (budgeting it) for the month.
- Hold for next month - It will move the money from the "To Budget" section of the header into the "For Next Month" section of the current month, and it will also show in the "Available Funds" of next month's header (if you have multiple months shown at once, it's easier to see this all on one page)  Essentially it's holding this month's income to be budgeted next month, effectively keeping it "out of sight, out of mind" for the current month so you don't fritter it away needlessly.
- Hold for all future months - a special case that really doesn't need to be used, and in fact has been mentioned that it will probably disappear at some point (don't worry about using it)
- Reset next months buffer- will pull money back from next month (that was put there by using "Hold for next month" above) to be available for budgeting this month, if you moved too much over, or some unexpected expense came up that you need to pull from the future.

As far as holding money over from next month, yes, that's the way it's designed.  It's based on the concept of "envelope budgeting" where you put $100 in an envelope for groceries.  At the end of the month, if you only spend $80, your envelope is still going to have $20 in it.  So, now you can decide, well, My monthly budget is $100, so I really only need to add $80 to catch up.  OR you can say, I don't mind building up a surplus, I'll add another $100, and now I've got $120.  If there's a special on chicken, I'll buy extra, 'cause I've got enough money.

**Comment by Mreco99** [comment]
Thats a big help thanks, I am used to envelope budgeting, and because my income and outgoings are random on any given month, I prefer to roll over left over budget instead of allocating it all that month. Otherwise if the 'leftover' bit is spread around some possibly deserving categories, I might need that somewhere else the next m month and have to go looking where I over budgeted. Just a personal tweak that works for me.
